sql_builder.go

```diff
@@ -50,7 +67,11 @@ var inRegex = regexp.MustCompile(`(?i)in\s*\(\s*\?\s*\)`)
 func (sq *sqlBuilder) compile() {
 	if sq.sql == "" {
 		if sq.Query.RawSQL.Fragment != "" {
-			sq.sql = sq.Query.RawSQL.Fragment
+			if sq.Query.Paginator != nil && !hasLimitOrOffset(sq.Query.RawSQL.Fragment) {
```

Comment:
Maybe I'm wrong there, but I think if you have `Paginator != nil` and `hasLimitOrOffset` is `true`, that should trigger a warning. If I asked for a pagination, but it doesn't paginate, it feels weird to fallback to the query **silently**.

Reply:
okay, in one of my project, i was using pagination from the start of query,
and somehow i transform query into sql
```
sql, args := query.ToSQL(&pop.Model{Value: structClass{}}, selectColumns...)
```
and then
```
and i use:
```
query.RawQuery(sql, args...)
```

i think i will test with tx.RawQuery instead for the second part.
will update soon

thank you

Reply:
If you're using the `Paginator` (so `Paginator != nil`), it means you're asking Pop to manage pagination, by adding the `LIMIT` and `OFFSET` statements.
My issue is, in the case you already finish the raw query with `LIMIT` and `OFFSET` statements (so `sq.Query.Paginator != nil && hasLimitOrOffset(sq.Query.RawSQL.Fragment)`), you fall into the else case, and use the raw query. That means the `Pagination` has no effect in this case (which is normal), but since the user asked for pagination on this query, it's not "working".

That's why I think you need to add a warning **for this specific case**, because I can't find a scenario where asking for pagination but providing the  `LIMIT` and `OFFSET` statements is not an error from the developer.